royale.go

```diff
@@ -10,58 +10,37 @@ type RoyaleRuleset struct {
 
 	Seed int64
 
+	// TODO: move Turn into BoardState?
 	Turn              int32
 	ShrinkEveryNTurns int32
-	DamagePerTurn     int32
-
-	// Output
-	OutOfBounds []Point
 }
 
 func (r *RoyaleRuleset) Name() string { return "royale" }
 
 func (r *RoyaleRuleset) CreateNextBoardState(prevState *BoardState, moves []SnakeMove) (*BoardState, error) {
-	if r.ShrinkEveryNTurns < 1 {
-		return nil, errors.New("royale game must shrink at least every turn")
+	if r.StandardRuleset.HazardDamagePerTurn < 1 {
+		return nil, errors.New("royale damage per turn must be greater than zero")
```

Comment:
This error was pulled out of the `damagePerTurn` method, since it's not true for StandardRuleset.
The `ShrinkEveryNTurns` check still exists inside of `populateHazards`.